Source/ACE.Server/WorldObjects/Hotspot.cs

```diff
@@ -129,26 +120,31 @@ private void Activate()
         private void Activate(Player plr)
         {
             var amount = DamageNext;
-            switch ((DamageType)_DamageType)
+            switch (DamageType)
             {
                 case DamageType.Mana:
-                    if (plr.Mana.Current >= plr.Mana.MaxValue) return;
-                    amount = (int)Math.Max(plr.Mana.Current + amount - plr.Mana.MaxValue, amount); // to-do: implement Player properties ManaMod using enchantment manager
-                    if (amount == 0) break;
-                    if (amount > 0) break; // to-do: top it off
-                    plr.Mana.Current -= (uint)amount;
-                    var msg = new GameMessagePrivateUpdateAttribute2ndLevel(plr, Vital.Mana, plr.Mana.Current);
-                    plr.Session.Network.EnqueueSend(msg);
+                    var manaDmg = amount;
+                    var result = plr.Mana.Current - manaDmg;
+                    if (result < 0 && manaDmg > 0)
+                        manaDmg += result;
+                    else if (result > plr.Mana.MaxValue && manaDmg < 0)
+                        manaDmg -= plr.Mana.MaxValue - result;
+                    if (manaDmg != 0)
+                    {
+                        plr.Mana.Current -= (uint)Math.Round(manaDmg);
+                        var msg = new GameMessagePrivateUpdateAttribute2ndLevel(plr, Vital.Mana, plr.Mana.Current);
+                        plr.Session.Network.EnqueueSend(msg);
```

Comment:
Please use plr.UpdateVital(plr.Mana, plr.Mana.Current - (uint)Math.Round(manaDmg)) to ensure vitals are being updated consistantly.